### ISSUE TITLE
[Fancy Block Placement] Ambient Occlusion Missing From Dummy Block & Weird Behaviors

### ISSUE THREAD

**Issue by Krutoy242** [Red-Studio-Ragnarok/Fancier-Block-Particles#12]
## Describe the bug

Placing blocks causing various rendering glitches, including:
1. Double of block on same place
  > For several frames, there is two blocks on same place - one moving and second stand still

3. Shadow blinking around block
  > When you place block, shadow appear for split second, then block "falls" without shadow, then shadow appear again when animation ends

5. Disappearing of block at last moment
  > Just frame before animation of block placement finished, block disappear, only frame of block visible

### Double block and and shadow blink:
![KqqZSYUxYW](https://user-images.githubusercontent.com/936618/209343312-11be7157-d936-401d-81dd-1aff845fa18d.gif)

### Transparent block at last moment
![javaw_OsVn6u8HlH](https://user-images.githubusercontent.com/936618/209343314-f541af53-3d90-4de4-a5a2-90140097e606.gif)

## Specification:

- OS: Windows 10
- CPU: i5-7400 @ 3.00 GHz
- GPU: GTX 1050 Ti
- RAM Allocated: 6GB

## Additional context

Happen on [E2E-E](https://www.curseforge.com/minecraft/modpacks/enigmatica-2-expert-extended) modpack.

**Comment by JustDesoroxxx** [comment]
Thanks for the report!

For the first point I am pretty sure this is fixed in the dev builds of FBP.

For the third, I need to talk with some peoples but I think this cannot be fixed (Actually I may have an idea, note to myself: Look into the materials property for the dummy block)

For the fifth point I never seen that behavior maybe it's also fixed in the dev build.

If you do not mind you could try with a test build and see if the issues are the same, they are posted on my Discord in the dev-builds channel (https://discord.gg/CBpqejmuWx)

I will release an up to date build right now.

**Comment by Krutoy242** [comment]
(oh, i just noticed, that when i added sub-descriptions GitHub incrememt numbers 😁)

**Comment by Krutoy242** [comment]
> If you do not mind you could try with a test build and see if the issues are the same, they are posted on my Discord in the dev-builds channel ([discord.gg/CBpqejmuWx](https://discord.gg/CBpqejmuWx))

Ok, i just tested dev build `Fancier-Block-Particles-0.8-Dev-9`. All three problems still there. Observing no changes.

**Comment by JustDesoroxxx** [comment]
Just came to mind, are you using Optifine?

**Comment by Krutoy242** [comment]
Yes, i do

**Comment by JustDesoroxxx** [comment]
Optifine might be causing this

**Comment by Krutoy242** [comment]
> Optifine might be causing this

I removed Optifine and tested again. All three problems still there, nothing fixed.

I used `0.8-Dev-9` version for test.

**Comment by JustDesoroxxx** [comment]
I could not reproduce the issue with any optimization mod I am suspecting either mod conflict or nvidia nvidiaing

**Comment by Krutoy242** [comment]
Then I could try binary search mod that causing it (duh). This could take week or two.

**Comment by JustDesoroxxx** [comment]
First thing try in an instance with only FBP to see if it's not nvidia nvidiaing

**Comment by Krutoy242** [comment]
Just tested, all three problems happen when barely no mods installed (only BoP and RecurrentComplex).

Through, problem with second invisible block not obivious, until i spammed zombies to create FPS lags.

![DyVnsIXwBS](https://user-images.githubusercontent.com/936618/210060828-94330962-11a0-4704-ab87-fe45d400040d.gif)

**Comment by JustDesoroxxx** [comment]
Yeah, this issue seems like a mix of Nvidia Nvidiaing and just me not being able to go faster than your CPU can go.

And putting the Fancy Block animation on a separate thread is probably impossible without breaking a bunch of mods

**Comment by JustDesoroxxx** [comment]
But I may be able to fix the shadow blinking.

But I am currently waiting Meldexun to make a pr of his big changes, since I don't want to cause more merge conflicts.

**Comment by Krutoy242** [comment]
> Nvidia Nvidiaing

Is that means, that "double block" and "transparent frame" will be issue for FBP for NVidia cards forever?

Nvidia is kinda popular type of video cards... I'm not sure whether to add this mod to the modpack, if it will glitch on all Nvidia cards 🤕

**Comment by JustDesoroxxx** [comment]
Well you can just turn of fancy block placement but from my understanding I currently cannot do much

**Comment by Krutoy242** [comment]
Then, I would appreciate if you not close this issue until its fixed, so I could keep subscribed to see when it fixed and then install on modpack.

P.S. I didnt knew that NVidia actually have more than 75% of market

![image](https://user-images.githubusercontent.com/936618/210171204-b608a11f-7c0a-4d76-8053-a2da8bdd0c63.png)

**Comment by JustDesoroxxx** [comment]
> Then, I would appreciate if you not close this issue until its fixed, so I could keep subscribed to see when it fixed and then install on modpack.
> 
> P.S. I didnt knew that NVidia actually have more than 75% of market
> 
> ![image](https://user-images.githubusercontent.com/936618/210171204-b608a11f-7c0a-4d76-8053-a2da8bdd0c63.png)

I will keep this open for the ambient occlusion missing problem (.3) but the rest I literally cannot fix at least if it's driver related.

**Comment by JustDesoroxxx** [comment]
> Then, I would appreciate if you not close this issue until its fixed, so I could keep subscribed to see when it fixed and then install on modpack.
> 
> P.S. I didnt knew that NVidia actually have more than 75% of market
> 
> ![image](https://user-images.githubusercontent.com/936618/210171204-b608a11f-7c0a-4d76-8053-a2da8bdd0c63.png)

I will keep this open for the ambient occlusion missing problem (.3) but the rest I literally cannot fix at least if it's driver related.

**Comment by JustDesoroxxx** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Project: Pacificus Ponens](https://trello.com/c/z0pwdSCv/48-project-pacificus-ponens)
